contracts/contracts/coordination/AllowList.sol

```diff
@@ -0,0 +1,53 @@
+pragma solidity ^0.8.0;
+import "@openzeppelin/contracts/access/AccessControlDefaultAdminRules.sol";
+import "./IAccessController.sol";
+import "./Coordinator.sol";
+
+
+contract AllowList is AccessControlDefaultAdminRules, IAccessController {
+    Coordinator public coordinator;
+
+    // mapp
+    mapping(uint256 => mapping(address => bool)) public rituals;
+
+    constructor(
+        Coordinator _coordinator,
+        address _admin
+    ) AccessControlDefaultAdminRules(0, _admin) {
+        coordinator = _coordinator;
+    }
+
+    function setCoordinator(Coordinator _coordinator) public {
+        require(hasRole(DEFAULT_ADMIN_ROLE, msg.sender), "Only admin can set coordinator");
+        coordinator = _coordinator;
+    }
+
+    function isEnricoAuthorized(
+        uint256 ritualID,
+        bytes memory evidence,
+        bytes memory ciphertextHash
+    ) public view override returns(bool) {
+        address enricoAddress = address(uint160(bytes20(evidence)));
```

Comment:
An address can't be the evidence, otherwise anyone can put any address there. The evidence was defined as a signature, so what we want here is to do something like:
`address enricoAddress = ecrecover(ciphertextHash, v, r, s)`
where `v, r, s` are somehow extracted from the signature `evidence`, and `ciphertextHash` should be somehow casted to bytes32 (what `ecrecover` expects). See https://docs.soliditylang.org/en/latest/cheatsheet.html#mathematical-and-cryptographic-functions

Reply:
@theref and I considered recovering the address on-chain via ECDSA, however in the case of the `AllowList` for example, the only way an address can be "authorized" is by `Ritual.authority` setting it in a permissioned way.  In other word, only the ritual's authority can add new enricos.  Are we missing something?

Reply:
Let's say the authority authorizes `0x1234`. I don't own that address, but I just write `0x1234` as evidence when creating the conditions and access control policy. Ursulas would accept this evidence and that's not what we want. We need to provide `evidence` that I own that address.

Reply:
If enricos are submitting a signature of the cipertext and policy, doesn't that mitigate the issue you indicate (as enrico cannot simply "write `0x1234` as evidence"?  

Ursula will recover the address from the signature provided in `AccessControlPolicy.authorization` and check that it's in the allow list.

Reply:
> If enricos are submitting a signature of the ciphertext and policy, doesn't that mitigate the issue you indicate (as enrico cannot simply "write `0x1234` as evidence")?

Yes, that's what I suggested initially.

> Ursula can recover the address from the signature provided in `AccessControlPolicy.authorization` and check that it's in the allow list.

Right, what I'm saying is that what you called `AccessControlPolicy.authorization` is what must be submitted as `evidence` by Ursula. That's what I originally meant by the evidence is attached to the ciphertext.

Reply:
Hmmm, not sure I'm following - Is there a problem with trusting ursula to perform a valid signature recovery?  Is that different that trusting ursula to perform a valid condition evaluation?

Reply:
Another way of framing it: What is the advantage of doing an authorization signature recovery onchian vs. in Ursula's runtime?

Reply:
The main advantage I see is that this allows for modular and pluggable evidence evaluation mechanisms (via smart contracts), without having to change Ursula's code